File: backend/src/main/java/com/yibang/erp/domain/entity/Warehouse.java
package com.yibang.erp.domain.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.time.LocalDateTime;

/**
 * 仓库实体类
 *
 * @author yibang-erp
 * @since 2024-01-14
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
@TableName("warehouses")
public class Warehouse {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /**
     * 仓库编码
     */
    @TableField("warehouse_code")
    private String warehouseCode;

    /**
     * 仓库名称
     */
    @TableField("warehouse_name")
    private String warehouseName;

    /**
     * 仓库类型：MAIN-主仓库，BRANCH-分仓库，TEMPORARY-临时仓库
     */
    @TableField("warehouse_type")
    private String warehouseType;

    /**
     * 所属公司ID
     */
    @TableField("company_id")
    private Long companyId;

    /**
     * 仓库地址
     */
    @TableField("address")
    private String address;

    /**
     * 联系人
     */
    @TableField("contact_person")
    private String contactPerson;

    /**
     * 联系电话
     */
    @TableField("contact_phone")
    private String contactPhone;

    /**
     * 仓库面积（平方米）
     */
    @TableField("area")
    private Double area;

    /**
     * 仓库状态：ACTIVE-启用，INACTIVE-停用，MAINTENANCE-维护中
     */
    @TableField("status")
    private String status;

    /**
     * 备注
     */
    @TableField("description")
    private String description;

    /**
     * 创建时间
     */
    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /**
     * 创建人ID
     */
    @TableField("created_by")
    private Long createdBy;

    /**
     * 更新人ID
     */
    @TableField("updated_by")
    private Long updatedBy;

    /**
     * 逻辑删除标记
     */
    @TableLogic
    @TableField("deleted")
    private Boolean deleted;
}

File: frontend/src/types/warehouse.ts
// 仓库实体类型
export interface Warehouse {
  id: number
  warehouseCode: string
  warehouseName: string
  warehouseType: 'MAIN' | 'BRANCH' | 'TEMPORARY'
  companyId: number
  address?: string
  contactPerson?: string
  contactPhone?: string
  area?: number
  status: 'ACTIVE' | 'INACTIVE' | 'MAINTENANCE'
  description?: string
  createdAt: string
  updatedAt: string
  createdBy: number
  updatedBy: number
  deleted: boolean
}

// 仓库请求类型
export interface WarehouseRequest {
  warehouseCode: string
  warehouseName: string
  warehouseType: 'MAIN' | 'BRANCH' | 'TEMPORARY'
  companyId: number
  address?: string
  contactPerson?: string
  contactPhone?: string
  area?: number
  status: 'ACTIVE' | 'INACTIVE' | 'MAINTENANCE'
  description?: string
}

// 仓库搜索参数类型
export interface WarehouseSearchParams {
  page: number
  size: number
  warehouseCode?: string
  warehouseName?: string
  warehouseType?: string
  status?: string
  companyId?: number
}

// 仓库类型选项
export const WAREHOUSE_TYPE_OPTIONS = [
  { label: '主仓库', value: 'MAIN' },
  { label: '分仓库', value: 'BRANCH' },
  { label: '临时仓库', value: 'TEMPORARY' }
]

// 仓库状态选项
export const WAREHOUSE_STATUS_OPTIONS = [
  { label: '启用', value: 'ACTIVE' },
  { label: '停用', value: 'INACTIVE' },
  { label: '维护中', value: 'MAINTENANCE' }
]
